feature_engine/estimator_checks.py

```diff
@@ -231,6 +238,17 @@ def check_get_feature_names_out(estimator):
                 "date2",
             ]
 
+            # user passes input features
+            features = ["var_3", "var_5", "var_7", "cat_var2", "date1"]
```

Comment:
I am not sure this is going to work, because when we set up the estimators, we are not defining the parameter `variables`. Also, we are making a test for all transformers, those that work with numerical and those that work with categorical, so we can't set the variables param universally for all. We need to think a bit what's the best way. What would you suggest?

Reply:
One way would be to create 3 global variables: 1) `_transformers_categorical`; 2) `_transformers_numerical`; and 3) `_transformers_categorical_and_numerical`.

I imagine that knowing whether a transformer is numerical or categorical will be important in other scenarios.

In the case of `check_get_feature_names_out`, we change the code to accommodate the 3 new global variables.

With regards to not defining the `variables` parameter, don't all of the classes create a `self.variables_` that correspond w/ the transformer - i.e., categorical, numerical, or both - when no `variables` parameter is provided when the class is instantiated?